runtime/src/main/java/org/corfudb/runtime/view/stream/StreamContext.java

```diff
@@ -0,0 +1,103 @@
+package org.corfudb.runtime.view.stream;
+
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.view.Address;
+
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import java.util.UUID;
+
+
+@ToString
+@Slf4j
+class StreamContext extends AbstractStreamContext {
+
+    /**
+     * A queue of addresses which have already been resolved.
+     */
+    final NavigableSet<Long> resolvedQueue
+            = new TreeSet<>();
+
+    /**
+     * The minimum global address which we have resolved this
+     * stream to.
+     */
+    long minResolution = Address.NON_ADDRESS;
+
+    /**
+     * The maximum global address which we have resolved this
+     * stream to.
+     */
+    long maxResolution = Address.NON_ADDRESS;
+
+    /**
+     * A priority queue of potential addresses to be read from.
+     */
+    final NavigableSet<Long> readQueue
+            = new TreeSet<>();
+
+    /**
+     * List of checkpoint records, if a successful checkpoint has been observed.
+     */
+    final NavigableSet<Long> readCpQueue = new TreeSet<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651470)